cloudomate/hoster/vps/clientarea.py

```diff
@@ -72,9 +72,12 @@ def _parse_service_row(self, row):
 
         status = columns[3].span.text.lower()
 
-        url = columns[4].a['href']
-        url = url.split('.php')
-        url = self._url + url[1]
+        try:
+            url = columns[4].a['href']
+            url = url.split('.php')
+            url = self._url + url[1]
+        except IndexError: #Fixes twosync
+            url = self._url + '/' + row['data-url']
```

Comment:
I don't think this is a good use case for exceptions. IMO, exceptions should be used for exceptional states, not for control flow. It's not clear which array can be out of bounds here, `columns` or `url`? It would be better e.g. to check the length of the array and respond accordingly.

Reply:
Agree